enterprise/backup/src/main/java/org/neo4j/backup/impl/BackupStrategyCoordinator.java

```diff
@@ -97,12 +103,27 @@ public void performBackup( OnlineBackupContext onlineBackupContext ) throws Comm
             causesOfFailure.forEach( commandFailed::addSuppressed );
             throw commandFailed;
         }
+        removeUnnecessaryTransactionLogs( destination.toFile(), outsideWorld.fileSystem() );
         if ( requiredArgs.isDoConsistencyCheck() )
         {
             performConsistencyCheck( onlineBackupContext.getConfig(), requiredArgs, consistencyFlags, destination );
         }
     }
 
+    private void removeUnnecessaryTransactionLogs( File backupDirectory, FileSystemAbstraction fileSystemAbstraction )
+    {
+        LifeSupport lifeSupport = new LifeSupport();
+        lifeSupport.add( forceCheckpointBackupService );
+
+        lifeSupport.init();
+        lifeSupport.start();
+        forceCheckpointBackupService.forceRotation();
+        forceCheckpointBackupService.forceCheckpoint();
+        forceCheckpointBackupService.forcePrune( );
+        lifeSupport.stop();
+        lifeSupport.shutdown();
```

Comment:
Do we really need a LifeSupport given that there's only one dependency?

Might be worth extracting the logic of init -> start -> (arbitrary Runnable) -> stop -> shutdown. Could be an instance method on LifeSupport.

Reply:
To clarify the above, great comments, thank you. We have decided to not force checkpoints/pruning manually and instead use configs. Therefore the above code doesn't exist anymore